web/src/email_pill.ts

```diff
@@ -8,24 +10,32 @@ type EmailPill = {
 
 export type EmailPillWidget = InputPillContainer<EmailPill>;
 
-const email_regex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
-
 export function create_item_from_email(
     email: string,
     current_items: EmailPill[],
 ): EmailPill | undefined {
-    if (!email_regex.test(email)) {
+    const original_email = email;
+    const parsed_address = parseOneAddress(email);
+    if (!parsed_address || parsed_address.type !== "mailbox") {
         return undefined;
     }
 
-    const existing_emails = current_items.map((item) => item.email);
-    if (existing_emails.includes(email)) {
+    const extracted_email = parsed_address.address;
+
+    const existing_emails = current_items
+        .map((item) => {
+            const parsed = parseOneAddress(item.email);
+            return parsed && parsed.type === "mailbox" ? parsed.address : null;
```

Comment:
Rather than parsing this here as well, can we store a field like `parsed_email` in the item that can be used to compare here?